### PR TITLE
fix: unnecessary store reinitialization in Socket module

### PR DESCRIPTION
**Pull request by danielmark0116**

### CODE REVIEW

File: lib/utils/socket.js [hunk 1]
```diff
@@ -202,12 +203,13 @@ class Socket {
   }
 
   subscribeStore = async () => {
-    return new Promise((resolve, reject) => {
-      const { configureStore } = this
-      const { store } = configureStore();
-      const unsubscribe = store.subscribe(() => {
-        resolve(store.getState()?.rooms?.isSocketFireOnInspectionUpdate)
-      })
+    return new Promise((resolve, _reject) => {
```
Comment (danielmark0116) [inline lib/utils/socket.js:206]: i think the `subscribe` is not needed? callback fired on the changed would resolve the promise either way, with the store value. also, calling configureStore was reinitializing the store.

I saw sth was not right in the redux debugger. Every time this fired, the redux history would go crazy

Comment (danielmark0116) [reply lib/utils/socket.js:206]: It was also doubling actions, saga effect where not working (like takeLatest etc)

File: lib/utils/socket.js [hunk 2]
```diff
@@ -72,6 +72,7 @@ class Socket {
     this.isEnableLF = get(options, 'isEnableLF', false);
     this.isEnableAudits = get(options, 'isEnableAudits', false);
     this.configureStore = get(options, 'configureStore', []);
+    this.store = get(options, 'store', {});
```
Comment (danielmark0116) [inline lib/utils/socket.js:75]: left the old option (`configureStore`) for backward compatibility

Comment (danielmark0116) [reply lib/utils/socket.js:75]: we would need to address this in other apps (currently I did the change only for the inspector)

https://github.com/Roomchecking/inspector/pull/53/files

**Review by monikap0612 — APPROVED** [review]

**Pull request merged**